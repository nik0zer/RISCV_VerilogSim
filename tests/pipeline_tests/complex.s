.section .text
.global _start

_start:
    addi x12, x0, 1
    add x1, x0, x12
    addi x5, x0, 10
    addi x6, x0, 5
    sub  x5, x5, x6
loop:
    addi x1, x1, 1
    sd x1, 0x10(x0)
    ld x1, 0x10(x0)
    beq x5, x1, endloop
    jal x10, loop
endloop:
    jal x4, poop
    addi x1, x1, 500
poop:
    addi x1, x0, 0
    slti x1, x5, 10
    addi x15, x0, 1
    beq x1, x15, end
    addi x1, x1, 600
end:
    addi x8, x0, 0
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
    nop
