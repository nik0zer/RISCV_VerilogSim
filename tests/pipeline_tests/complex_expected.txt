x
x
x
x
0000000000000001
0000000000000001
000000000000000a
0000000000000005
0000000000000005
0000000000000002
x
0000000000000002
x
x
0x0000000000010028
x
x
0000000000000003
x
0000000000000003
x
x
0x0000000000010028
x
x
0000000000000004
x
0000000000000004
x
x
0x0000000000010028
x
x
0000000000000005
x
0000000000000005
x
x
x
x
0x000000000001002c
x
x
0x0000000000000000
0x0000000000000001
0x0000000000000001
x
x
x
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000
000000000000000000